script.service.janitor/viewer.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+import utils
+from xbmcgui import Dialog, WindowXMLDialog
+
+
+class LogViewerDialog(WindowXMLDialog):
+    """
+    The LogViewerDialog class is an extension of the default windows supplied with Kodi.
+
+    It is used to display the contents of a log file, and as such uses a fullscreen window to show as much text as
+    possible. It also contains two buttons for trimming and clearing the contents of the log file.
+    """
+    TEXTBOXID = 202
+    TRIMBUTTONID = 301
+    CLEARBUTTONID = 302
+
+    def __init__(self, xml_filename, script_path, default_skin="Default", default_res="720p", *args, **kwargs):
+        self.log = utils.Log()
+        WindowXMLDialog.__init__(self)
```

Comment:
[Documentation](https://codedocs.xyz/xbmc/xbmc/group__python__xbmcgui__window__dialog__xml.html)  tells that `xmlFilename` and `scriptPath` are required arguments for this function. Does it work correctly without passing those arguments?
```suggestion
        WindowXMLDialog.__init__(self, xml_filename, script_path)
```

Reply:
It's always worked for me this way, even when developing the new log file viewer based on the Estuary skin the window opened just fine everytime.